src/sentry/mediators/sentry_apps/creator.py

```diff
@@ -75,6 +78,21 @@ def _create_ui_components(self):
                 schema=element,
             )
 
+    def _create_integration_feature(self):
+        # sentry apps must have at least one feature
+        # defaults to 'integrations-api'
+        try:
```

Comment:
This is the wrong pattern. It also needs to be wrapped ina  transaction block.

```python
try:
    with transaction.atomic():
        IntegrationFeature.objects.create(...)
except IntegrityError:
    pass
```

And do you really need to log when this happens? This behavior is more like, "ensure this exists, but fail silently" kinda thing when we do it elsewhere. We never log this.

Reply:
This whole object's logic is run inside of a transaction. Every Mediators' is. https://github.com/getsentry/sentry/blob/master/src/sentry/mediators/mediator.py#L156-L164

Reply:
The whole mediator is wrapped in a transaction, so is it necessary to have it within that specific block too? 

Also since this is in the creator, there really shouldn't be one that already exists - so if that does happen maybe I'd want to know?

Reply:
> The whole mediator is wrapped in a transaction, so is it necessary to have it within that specific block too?

Yes, it still needs a transaction. Technically a checkpoint. Otherwise, it'll break your other transaction and put it in an invalid state.

> Also since this is in the creator, there really shouldn't be one that already exists - so if that does happen maybe I'd want to know?

That's your call, and just depends or your expectations.